src/NetAnalyzers/CSharp/Microsoft.NetCore.Analyzers/InteropServices/CSharpDisableRuntimeMarshalling.Fixer.cs

```diff
@@ -0,0 +1,119 @@
+                            equivalenceKey: nameof(MicrosoftNetCoreAnalyzersResources.UseDisabledMarshallingEquivalent)),
+                        diagnostic);
+                }
+            }
+        }
+
+        private static async Task<Document> UseDisabledMarshallingEquivalentAsync(SyntaxNode node, Document document, CancellationToken ct)
+        {
+            var editor = await DocumentEditor.CreateAsync(document, ct).ConfigureAwait(false);
+            var operation = (IInvocationOperation)editor.SemanticModel.GetOperation(node, ct);
+            InvocationExpressionSyntax syntax = (InvocationExpressionSyntax)operation.Syntax;
+            var enclosingMethod = FindEnclosingMethod(syntax);
+            bool addUnsafeToEnclosingMethod = false;
+
+            if (operation.TargetMethod.Name == "SizeOf")
+            {
+                if (operation.TargetMethod.IsGenericMethod)
+                {
+                    if (operation.TargetMethod.TypeArguments[0].IsUnmanagedType)
+                    {
+                        addUnsafeToEnclosingMethod = true;
+                        editor.ReplaceNode(syntax, SyntaxFactory.SizeOfExpression(GetGenericTypeParameterSyntax(syntax)));
+                    }
+                }
+                else if (operation.Arguments[0].Value is ITypeOfOperation { TypeOperand.IsUnmanagedType: true } typeOf)
+                {
+                    addUnsafeToEnclosingMethod = true;
+                    editor.ReplaceNode(syntax, SyntaxFactory.SizeOfExpression(GetTypeOfTypeSyntax((TypeOfExpressionSyntax)typeOf.Syntax)));
+                }
+            }
+            if (operation.TargetMethod.Name == "StructureToPtr" && operation.Arguments[0].Value.Type.IsUnmanagedType)
+            {
+                addUnsafeToEnclosingMethod = true;
+                editor.ReplaceNode(syntax,
+                    editor.Generator.AssignmentStatement(
+                        SyntaxFactory.PrefixUnaryExpression(SyntaxKind.PointerIndirectionExpression,
+                            (ExpressionSyntax)editor.Generator.CastExpression(editor.SemanticModel.Compilation.CreatePointerTypeSymbol(operation.Arguments[0].Value.Type),
+                                operation.Arguments[1].Value.Syntax)),
+                        operation.Arguments[0].Value.Syntax));
+            }
+
+            if (addUnsafeToEnclosingMethod)
+            {
+                editor.SetModifiers(enclosingMethod, editor.Generator.GetModifiers(enclosingMethod).WithIsUnsafe(true));
+            }
+
+            return editor.GetChangedDocument();
+
+            static TypeSyntax GetGenericTypeParameterSyntax(InvocationExpressionSyntax syntax)
+            {
+                var memberAccess = (MemberAccessExpressionSyntax)syntax.Expression;
+                var genericName = (GenericNameSyntax)memberAccess.Name;
```

Comment:
Is it possible the expression bind against the generic method, but the `<..>` is omitted from syntax and it's implicitly inferred, in which case the cast could fail?